controller/node_controller.go

```diff
@@ -995,6 +1001,139 @@ func (nc *NodeController) syncNodeStatus(pod *corev1.Pod, node *longhorn.Node) e
+	if err != nil {
+		node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeMultipathd, longhorn.ConditionStatusFalse,
+			string(longhorn.NodeConditionReasonNamespaceExecutorErr),
+			fmt.Sprintf("Failed to get namespace executor: %v", err.Error()))
+		return
+	}
+	args := []string{"show", "status"}
+	if result, _ := nsexec.Execute(nil, "multipathd", args, lhtypes.ExecuteDefaultTimeout); result != "" {
+		node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeMultipathd, longhorn.ConditionStatusFalse,
+			string(longhorn.NodeConditionReasonMultipathdIsRunning),
+			"multipathd is running with a known issue that affects Longhorn. See description and solution at https://longhorn.io/kb/troubleshooting-volume-with-multipath")
+		return
+	}
+
+	node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeMultipathd, longhorn.ConditionStatusTrue, "", "")
+}
+
+func (nc *NodeController) syncNFSClientVersion(kubeNode *corev1.Node, node *longhorn.Node, namespaces []lhtypes.Namespace) {
+	kernelVersion := kubeNode.Status.NodeInfo.KernelVersion
+	nfsClientVersions := []string{"CONFIG_NFS_V4_2", "CONFIG_NFS_V4_1", "CONFIG_NFS_V4"}
+
+	nsexec, err := lhns.NewNamespaceExecutor(lhtypes.ProcessNone, lhtypes.HostProcDirectory, namespaces)
+	if err != nil {
+		node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeNFSClientInstalled, longhorn.ConditionStatusFalse,
+			string(longhorn.NodeConditionReasonNamespaceExecutorErr),
+			fmt.Sprintf("Failed to get namespace executor: %v", err.Error()))
+		return
+	}
+
+	kernelConfigPath := "/boot/config-" + kernelVersion
+	args := []string{kernelConfigPath}
+	if _, err := nsexec.Execute(nil, "ls", args, lhtypes.ExecuteDefaultTimeout); err != nil {
+		node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeNFSClientInstalled, longhorn.ConditionStatusFalse,
+			string(longhorn.NodeConditionReasonKernelConfigIsNotFound),
+			fmt.Sprintf("Failed to check %v, because %v does not exist: %v", nfsClientVersions, kernelConfigPath, err.Error()))
+		return
+	}
+
+	for _, ver := range nfsClientVersions {
+		args := []string{ver + "=", kernelConfigPath}
+		result, err := nsexec.Execute(nil, "grep", args, lhtypes.ExecuteDefaultTimeout)
+		if err != nil {
+			nc.logger.WithError(err).Debugf("Failed to find kernel config %v on node %v", ver, node.Name)
+			continue
+		}
+		enabled := strings.TrimSpace(strings.Split(result, "=")[1])
+		switch enabled {
+		case "y":
+			node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeNFSClientInstalled, longhorn.ConditionStatusTrue, "", "")
+			return
+		case "m":
+			opts := []string{"|grep", "nfs"}
+			if _, err := nsexec.Execute(nil, "lsmod", opts, lhtypes.ExecuteDefaultTimeout); err == nil {
```

Comment:
What happens if `lsmod` is no found in the host side?
Can we install `kmod` in longhorn-manager and check the modules inside the pod?

Reply:
>What happens if lsmod is no found in the host side?

An error string "No such file or directory" or "Command 'lsmod' not found" in returning `err`.

>nit: can we install kmod in longhorn-manager and check the modules inside the pod? we can improve it later.

Yes, `kmod` can get the module status of the worker node in the longhorn-manager pod.

Reply:
Modified to use 'kmod' command in the longhorn-manager pod.